add examples of `mandatory` and clarify its description (#139)
openid/OpenID4VCI: openid-4-verifiable-credential-issuance-1_0.md

* `credential_definition`: REQUIRED. Object containing the detailed description of the Credential type. It consists at least of the following two sub claims:
  * `type`: REQUIRED. Array designating the types a certain Credential type supports according to [@VC_DATA], Section 4.3.
  * `credentialSubject`: OPTIONAL. An object containing a list of name/value pairs, where each name identifies a claim offered in the Credential. The value can be another such object (nested data structures), or an array of such objects. To express the specifics about the claim, the most deeply nested value MAY be an object that includes a following non-exhaustive list of parameters defined by this specification:

>       * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If the `mandatory` property is omitted its default should be assumed to be `false`.

Reviewer: what about `mandatory` -> `required`?

Reviewer: I find the parameter `required` used in other context and more recognizable than `mandatory`

Author: I don't have a strong opinion here.. suggest we bikeshed in the WG call/this PR comment.

Reviewer 2: I think mandatory is better language because its the issuer saying the claim is mandatory in this credential they issue. "required" would kind of be a requirement the issuer is placing on its self in this context which makes less sense to me.

Reviewer 3: Discussed on today's editors and no one felt required was significantly better, and I kind of agree with Tobias that mandatory is slightly better for this case.

Reviewer 4: I can live with either, and I think 'mandatory' is just fine.

Author: @peppelinux I think we are sticking with `mandatory`

>       * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If the `mandatory` property is omitted its default should be assumed to be `false`.

Reviewer 5: ```suggestion
      * `mandatory`: OPTIONAL. Boolean which, when set to `true`, it signifies that the claim is always included in the issued Credential. If set to `false`, the Issuer can choose not to include the claim in certain cases. If the `mandatory` parameter is omitted, the default value is `false`.
```

Author: ```suggestion
      * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the Credential Issuer can choose not to include the claim. If the `mandatory` parameter is omitted, the default should be assumed to be `false`.
```

Reviewer 3: "the Credential Issuer can choose not to include the claim" potentially seems to suggesting the decision is only made by the issuer; I was presuming the decision could also be made by the user or based on what the wallet requests?

Perhaps something more like "the Credential Issuer may issue Credentials that do not include the claim" ?

Reviewer 5: That is why I thought we would need to be more precise in the first place. If the issuer does not have the info, the issuer cannot return it and the wallet should not fail.

Reviewer 5: ```suggestion
 * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the claim is not included in the issued Credential if the wallet did not request the inclusion of the claim in the Authorization Request, or if the Issuer chose to not include the claim. If the `mandatory` parameter is omitted, the default should be assumed to be `false`.
```

>       * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the Credential Issuer can choose not to include the claim. If the `mandatory` parameter is omitted, the default should be assumed to be `false`.

Reviewer 5: ```suggestion
      * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the claim is not included in the issued Credential if the wallet did not request the inclusion of the claim in the Authorization Request, or if the Issuer chose to not include the claim. If the `mandatory` parameter is omitted, the default should be assumed to be `false`.
```

>       * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the claim is not included in the issued Credential if the wallet did not request the inclusion of the claim in the Authorization Request, or if the Issuer chose to not include the claim. If the `mandatory` parameter is omitted, the default should be assumed to be `false`.

Author: ```suggestion
      * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the claim is not included in the issued Credential if the wallet did not request the inclusion of the claim, or if the Issuer chose to not include the claim. If the `mandatory` parameter is omitted, the default should be assumed to be `false`.
```

Author: wallet can use credential request or authorization details in the token request to narrow down this too, so strictly speaking it is not only authz req.
I also worry that the current/proposed language implies that if the wallet requests the issuer will have to issue the credential.

  * `@context`: REQUIRED. Array as defined in [@VC_DATA], Section 4.1.
  * `type`: REQUIRED. Array designating the types a certain credential type supports according to [@VC_DATA], Section 4.3.
  * `credentialSubject`: OPTIONAL. An object containing a list of name/value pairs, where each name identifies a claim offered in the Credential. The value can be another such object (nested data structures), or an array of such objects. To express the specifics about the claim, the most deeply nested value MAY be an object that includes a following non-exhaustive list of parameters defined by this specification:

>       * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If the `mandatory` property is omitted its default should be assumed to be `false`.

Reviewer 5: ```suggestion
      * `mandatory`: OPTIONAL. Boolean which, when set to `true`, it signifies that the claim is always included in the issued Credential. If set to `false`, the Issuer can choose not to include the claim in certain cases. If the `mandatory` parameter is omitted, the default value is `false`.
```

Author: ```suggestion
      * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the Credential Issuer can choose not to include the claim. If the `mandatory` parameter is omitted, the default should be assumed to be `false`.
```

>       * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the Credential Issuer can choose not to include the claim. If the `mandatory` parameter is omitted, the default should be assumed to be `false`.

Reviewer 3: "the Credential Issuer can choose not to include the claim" potentially seems to suggesting the decision is only made by the issuer; I was presuming the decision could also be made by the user or based on what the wallet requests?

Perhaps something more like "the Credential Issuer may issue Credentials that do not include the claim" ?

```suggestion
      * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the Credential Issuer may issue Credentials that do not include the claim. If the `mandatory` parameter is omitted, the default should be assumed to be `false`.
```

Reviewer 5: This might be more accurate:

```suggestion
 * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the claim is not included in the issued Credential if the wallet did not request the inclusion of the claim in the Authorization Request, or if the Issuer chose to not include the claim. If the `mandatory` parameter is omitted, the default should be assumed to be `false`.
```

>  * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the claim is not included in the issued Credential if the wallet did not request the inclusion of the claim in the Authorization Request, or if the Issuer chose to not include the claim. If the `mandatory` parameter is omitted, the default should be assumed to be `false`.

Author: ```suggestion
 * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the claim is not included in the issued Credential if the wallet did not request the inclusion of the claim, or if the Issuer chose to not include the claim. If the `mandatory` parameter is omitted, the default should be assumed to be `false`.
```

* `doctype`: REQUIRED. String identifying the Credential type as defined in [@!ISO.18013-5].
* `claims`: OPTIONAL. An object containing a list of name/value pairs, where the name is a certain `namespace` as defined in [@!ISO.18013-5] (or any profile of it), and the value is an object. This object also contains a list of name/value pairs, where the name is a claim name value that is defined in the respective namespace and is offered in the Credential. The value is an object detailing the specifics of the claim with the following non-exhaustive list of parameters that MAY be included:

>     * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If the `mandatory` property is omitted its default should be assumed to be `false`.

Reviewer 5: ```suggestion
    * `mandatory`: OPTIONAL. Boolean which, when set to `true`, it signifies that the claim is always included in the issued Credential. If set to `false`, the Issuer can choose not to include the claim in certain cases. If the `mandatory` parameter is omitted, the default value is `false`.
```

Author: ```suggestion
    * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the Credential Issuer can choose not to include the claim. If the `mandatory` parameter is omitted, the default should be assumed to be `false`.
```

>     * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the claim is not included in the issued Credential if the wallet did not request the inclusion of the claim in the Authorization Request, or if the Issuer chose to not include the claim. If the `mandatory` parameter is omitted, the default should be assumed to be `false`.

Author: ```suggestion
    * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the claim is not included in the issued Credential if the wallet did not request the inclusion of the claim, and/or if the Credential Issuer chose to not include the claim. If the `mandatory` parameter is omitted, the default should be assumed to be `false`.
```
